python/pyarrow/includes/libarrow.pxd

```diff
@@ -81,6 +81,11 @@ cdef extern from "arrow/config.h" namespace "arrow" nogil:
     CRuntimeInfo GetRuntimeInfo()
 
 
+cdef extern from "arrow/util/future.h" namespace "arrow" nogil:
+    cdef cppclass CFuture_Void" arrow::Future<>":
+        CStatus status()
```

Comment:
Surprising we've made it this far without exposing `Future`.  I wonder if we want to add a `WaitForFinish` method on ExecPlan instead of exposing `Future` (essentially a sync mirror-api) like we do in the `Scanner`.

I don't really have any problems with this approach though.  Just making an observation.

Reply:
I suppose at some point we'll want to expose async APIs in Python, so we're probably bound to interface with `Future` in Cython anyway.